honeybee_energy/constructionset.py

```diff
@@ -0,0 +1,801 @@
+        self._interior_construction = value
+
+    @property
+    def ground_construction(self):
+        """Get or set the OpaqueConstruction for ground-contact floor slabs."""
+        if self._ground_construction is None:
+            return generic_ground_slab
+        return self._ground_construction
+
+    @ground_construction.setter
+    def ground_construction(self, value):
+        if value is not None:
+            self._check_construction(value)
+        self._ground_construction = value
+
+    def __repr__(self):
+        return 'Floor Construction Set:\n Exterior: {}\n Interior: {}' \
+            '\n Ground: {}'.format(self.exterior_construction.name,
+                                   self.interior_construction.name,
+                                   self.ground_construction.name)
+
+
+class RoofCeilingSet(_FaceSetBase):
+    """Set containing all energy constructions needed to for an energy model's Roofs.
+
+    Properties:
+        exterior_construction
+        interior_construction
+        ground_construction
+        is_default
+    """
+    __slots__ = ()
+
+    @property
+    def exterior_construction(self):
+        """Get or set the OpaqueConstruction for exterior roofs."""
+        if self._exterior_construction is None:
+            return generic_roof
+        return self._exterior_construction
+
+    @exterior_construction.setter
+    def exterior_construction(self, value):
+        if value is not None:
+            self._check_construction(value)
+        self._exterior_construction = value
+
+    @property
+    def interior_construction(self):
+        """Get or set the OpaqueConstruction for interior ceilings."""
+        if self._interior_construction is None:
+            return generic_interior_ceiling
+        return self._interior_construction
+
+    @interior_construction.setter
+    def interior_construction(self, value):
+        if value is not None:
+            self._check_construction(value)
+        self._interior_construction = value
+
+    @property
+    def ground_construction(self):
+        """Get or set the OpaqueConstruction for underground roofs."""
+        if self._ground_construction is None:
+            return generic_underground_roof
+        return self._ground_construction
+
+    @ground_construction.setter
+    def ground_construction(self, value):
+        if value is not None:
+            self._check_construction(value)
+        self._ground_construction = value
+
+    def __repr__(self):
+        return 'RoofCeiling Construction Set:\n Exterior: {}\n Interior: {}' \
+            '\n Ground: {}'.format(self.exterior_construction.name,
+                                   self.interior_construction.name,
+                                   self.ground_construction.name)
+
+
+class ApertureSet(object):
+    """Set containing all energy constructions needed to for an energy model's Apertures.
+
+    Properties:
+        fixed_window_construction
+        interior_construction
```

Comment:
I disagree because the interior_construction gets assigned to all interior apertures, whether they are interior 'glass_door` or interior `operable_window`, which is how people want it 99% of the time (you really only care about different constructions for different aperture types when they are on the exterior).  You can see that I clarify this in the docstrings and, if this inconsistency caught your attention, then it had the intended effect of trying to make you aware of this.